pilot/pkg/serviceregistry/kube/controller/ambientindex.go

```diff
@@ -737,11 +742,11 @@ func (a *AmbientIndex) handlePod(oldObj, newObj any, isDelete bool, c *Controlle
 		if oldWl != nil {
 			// If we already knew about this workload, we need to make sure we drop all VIP references as well
 			for vip := range oldWl.VirtualIps {
-				a.dropWorkloadFromService(vip, p.Status.PodIP)
+				a.dropWorkloadFromService(vip, oldWl.ResourceName())
 			}
 			log.Debugf("%v: workload removed, pushing", p.Status.PodIP)
 			// TODO: namespace for network?
-			updates.Insert(model.ConfigKey{Kind: kind.Address, Name: p.Status.PodIP})
+			updates.Insert(model.ConfigKey{Kind: kind.Address, Name: oldWl.ResourceName()})
```

Comment:
TODO on L748 is addressed so should be removed.

However, the TODO recommended making the key

```
{
kind: address
name: IP
namespace: network
}

instead of `name: network/ip`. Can we do that? That should avoid some splitting/joining?